### ISSUE TITLE
mvcc commits everything after restart, even not finished transactions

### ISSUE THREAD

**Issue by Gerold103** [tarantool/tarantool#5973]
When MVCC is enabled, it works fine until restart. If some transactions after restart are not committed (synchronous ones waiting for CONFIRM), they become visible anyway. Here is a test:
```Lua
fiber = require('fiber')
box.cfg{
    memtx_use_mvcc_engine = true,
    replication_synchro_quorum = 2,
    replication_synchro_timeout = 1000
}
s = box.schema.create_space('test', {is_sync = true})
_ = s:create_index('pk')
lsn = box.info.lsn
fiber.new(function() s:replace{1} end)
while box.info.lsn == lsn do fiber.sleep(0.1) end
assert(s:get{1} == nil) -- Passes.
os.exit(0)


box.cfg{
    memtx_use_mvcc_engine = true,
    replication_synchro_quorum = 2,
    replication_synchro_timeout = 1000
}
s = box.space.test
assert(s:get{1} == nil) -- Fails!!!
```
The transaction is not committed. It is still in the limbo, `txn_complete_success()` wasn't called yet. You can see it in the debugger - the limbo queue is not empty (has 1 element).

I see that MVCC is used in memtx only by `memtx_space_replace_all_keys()`, which works after recovery is finished. During recovery it is never used. That might be the problem.

**Comment by alyapunov** [comment]
Duplicate of #5610. Will be fixed in #5610.

**Comment by alyapunov** [comment]
Duplicate of #5610
